diff --git a/scripts/travis-tool.sh b/scripts/travis-tool.sh
index f87ce1f..f1c1161 100755
--- a/scripts/travis-tool.sh
+++ b/scripts/travis-tool.sh
@@ -11,6 +11,9 @@ BIOC=${BIOC:-"http://bioconductor.org/biocLite.R"}
 BIOC_USE_DEVEL=${BIOC_USE_DEVEL:-"TRUE"}
 OS=$(uname -s)
 
+## Possible drat repos, unset by default
+DRAT_REPOS=${DRAT_REPOS:-""}
+
 PANDOC_VERSION='1.13.1'
 PANDOC_DIR="${HOME}/opt/pandoc"
 PANDOC_URL="https://s3.amazonaws.com/rstudio-buildtools/pandoc-${PANDOC_VERSION}.zip"
@@ -44,6 +47,19 @@ Bootstrap() {
     if ! (test -e .Rbuildignore && grep -q 'travis-tool' .Rbuildignore); then
         echo '^travis-tool\.sh$' >>.Rbuildignore
     fi
+
+    SetRepos
+}
+
+SetRepos() {
+    echo "local({" >> ~/.Rprofile
+    echo "   r <- getOption(\"repos\");" >> ~/.Rprofile
+    echo "   r[\"CRAN\"] <- \"${CRAN}\"" >> ~/.Rprofile
+    for d in ${DRAT_REPOS}; do 
+        echo "   r[\"${d}\"] <- \"http://${d}.github.io/drat\"" >> ~/.Rprofile
+    done        
+    echo "   options(repos=r)" >> ~/.Rprofile
+    echo "})" >> ~/.Rprofile
 }
 
 InstallPandoc() {
@@ -74,7 +90,7 @@ BootstrapLinux() {
     # Install an R development environment. qpdf is also needed for
     # --as-cran checks:
     #   https://stat.ethz.ch/pipermail/r-help//2012-September/335676.html
-    Retry sudo apt-get install --no-install-recommends r-base-dev r-recommended qpdf
+    Retry sudo apt-get install -y --no-install-recommends r-base-dev r-recommended qpdf
 
     # Change permissions for /usr/local/lib/R/site-library
     # This should really be via 'staff adduser travis staff'
@@ -90,7 +106,7 @@ BootstrapLinuxOptions() {
         # We add a backports PPA for more recent TeX packages.
         sudo add-apt-repository -y "ppa:texlive-backports/ppa"
 
-        Retry sudo apt-get install --no-install-recommends \
+        Retry sudo apt-get install -y --no-install-recommends \
             texlive-base texlive-latex-base texlive-generic-recommended \
             texlive-fonts-recommended texlive-fonts-extra \
             texlive-extra-utils texlive-latex-recommended texlive-latex-extra \
@@ -152,7 +168,7 @@ AptGetInstall() {
     fi
 
     echo "Installing apt package(s) $@"
-    Retry sudo apt-get install "$@"
+    Retry sudo apt-get -y install "$@"
 }
 
 DpkgCurlInstall() {
@@ -182,7 +198,7 @@ RInstall() {
     fi
 
     echo "Installing R package(s): $@"
-    Rscript -e 'install.packages(commandArgs(TRUE), repos="'"${CRAN}"'")' "$@"
+    Rscript -e 'install.packages(commandArgs(TRUE))' "$@"
 }
 
 BiocInstall() {
@@ -219,12 +235,12 @@ InstallGithub() {
 
     echo "Installing GitHub packages: $@"
     # Install the package.
-    Rscript -e 'library(devtools); library(methods); options(repos=c(CRAN="'"${CRAN}"'")); install_github(commandArgs(TRUE), build_vignettes = FALSE)' "$@"
+    Rscript -e 'library(devtools); library(methods); install_github(commandArgs(TRUE), build_vignettes = FALSE)' "$@"
 }
 
 InstallDeps() {
     EnsureDevtools
-    Rscript -e 'library(devtools); library(methods); options(repos=c(CRAN="'"${CRAN}"'")); install_deps(dependencies = TRUE)'
+    Rscript -e 'library(devtools); library(methods); install_deps(dependencies = TRUE)'
 }
 
 InstallBiocDeps() {
